contextproject-ms3/src/main/java/contextproject/audio/TrackProcessor.java

```diff
@@ -148,15 +179,12 @@ private void setupDispatcherChain(double startGain, double startBpm) throws Enco
     gainProcessor = new GainProcessor(startGain);
     dispatcher = new CustomAudioDispatcher(tarsosStream, wsola.getInputBufferSize(),
         wsola.getOverlap());
+    progressProcessor = new ProgressProcessor();
+    progressProcessor.setUp(track.getDuration(), this.secondsToSkip, pcc);
```

Comment:
I think it's better to remove the setUp method and move all of this into the constructor.